spanner/google/cloud/spanner/snapshot.py

```diff
@@ -92,17 +90,20 @@ def read(self, table, columns, keyset, index='', limit=0,
         iterator = api.streaming_read(
             self._session.name, table, columns, keyset.to_pb(),
             transaction=transaction, index=index, limit=limit,
-            resume_token=resume_token, options=options)
+            options=options)
 
         self._read_request_count += 1
 
+        restart = functools.partial(
+            api.streaming_read, self._session.name, table, columns, keyset,
```

Comment:
you need to also pass in the transaction here.

Reply:
Should it pass `options` too?

Reply:
I guess so. What does `options` have?

Reply:
8f47fe2

Reply:
Options contains the metadata prefix (with the database name).